pkg/storage/chunk/cache/memcached.go

```diff
@@ -94,7 +106,12 @@ func NewMemcached(cfg MemcachedConfig, client MemcachedClient, name string, reg
 					batchID: input.batchID,
 				}
 				res.found, res.bufs, res.missed, res.err = c.fetch(input.ctx, input.keys)
-				input.resultCh <- res
+				// NOTE: This check is needed because goutines submitting work via `inputCh` may exit in-between because of context cancellation or timeout. This helps to close these worker goroutines to exit without hanging around.
+				select {
+				case <-c.closed:
+					return
```

Comment:
This is not covered by a test, is it worth adding one?

Reply:
We are actually in-directly testing this, as without this, some of the worker go-routine may hang on `resultsCh` so that not all `wg.Done()` is run so `wg.Wait()` on the `Stop()` method will not finish.

We in-directly testing this by explicitly calling `m.Stop()` after every sub tests.

Hard to test this specific line deterministically in test if that's what you mean also tbh not necessary I think (testing `Stop()` is good enough IMHO)